### ISSUE TITLE
AsciidocCaptureModule wrong decorate ManyRelationshipRepository

### ISSUE THREAD

**Issue by 0ffer** [crnk-project/crnk-framework#532]
When I am try to get ManyRelationshipRepository from CrnkClient that wrapped with AsciidocCaptureModule I get an error 
```
java.lang.ClassCastException: class io.crnk.gen.asciidoc.capture.AsciidocCaptureModule$AsciidocOneRelationshipDecorator cannot be cast to class io.crnk.core.repository.ManyRelationshipRepository (io.crnk.gen.asciidoc.capture.AsciidocCaptureModule$AsciidocOneRelationshipDecorator and io.crnk.core.repository.ManyRelationshipRepository are in unnamed module of loader 'app')
```

I found, that AsciidocCaptureModule wraps repositories not correctly.
The module contains the next code:
```java
@Override
        public Object decorateRepository(Object repository) {
            if (repository instanceof ResourceRepository) {
                return new AsciidocResourceDecorator((ResourceRepository) repository);
            }
            if (repository instanceof OneRelationshipRepository) {
                return new AsciidocOneRelationshipDecorator((OneRelationshipRepository) repository);
            }
            if (repository instanceof ManyRelationshipRepository) {
                return new AsciidocManyRelationshipDecorator((ManyRelationshipRepository) repository);
            }
            return repository;
        }
``` 
And when it runs the next repository type come in:

```java
@Deprecated
public interface RelationshipRepository<T, I, D, J>
        extends Repository, MatchedRelationshipRepository,
        ManyRelationshipRepository<T, I, D, J>, OneRelationshipRepository<T, I, D, J>
```

Now it is impossible to get correctly decorated ManyRelationshipRepository because if-statement always stopped on OneRelationshipRepository.

**Comment by remmeier** [comment]
I will do the 3.0 release these days. Should you have time to look at  this, let me know. Or otherwise I can take a quick look before the release.

**Comment by 0ffer** [comment]
> I will do the 3.0 release these days. Should you have time to look at this, let me know. Or otherwise I can take a quick look before the release.

Yes, I will try to fix it. I already forked repository and saw the problem code.

**Comment by 0ffer** [comment]
Fixed with PR#538
